server/models/element.ts

```diff
@@ -53,13 +65,29 @@ export function updateElementDoc (newAttributes, oldElement) {
     if (!attributes.updatable.includes(attribute)) {
       throw newError('invalid attribute', 400, { attribute, oldElement })
     }
-    const newVal = newAttributes[attribute]
+    let newVal
+    if (attribute === 'ordinal') {
+      newVal = findNewOrdinal(oldElement, listingElements, newAttributes[attribute])
+    } else {
+      newVal = newAttributes[attribute] || defaultValues[attribute]?.()
+      validations.pass(attribute, newVal)
+    }
+    if (newVal !== undefined) {
```

Comment:
[non-blocking]
The convention within this code base is to check against both `null` and `undefined`, with this Douglas Crockford-approved exception use of the non-strict-egal comparison:
 ```suggestion
     if (newVal != null) {
```
which is strictly equivalent to 
 ```js
    if (newVal !== undefined && newVal !== null) {
```

Reply:
done in 08fdfab86